### PR TITLE
Add Gitleaks workflow

### LINKED ISSUE #23: workflow to detect/prevent hardcoded secrets in git repos
- Use [gitleaks](https://github.com/zricethezav/gitleaks) to detect and prevent hardcoded secrets (e.g., passwords, api keys, and tokens) in git repos
- It would be great if we could run gitleaks as a [pre-commit hook](https://pre-commit.com/) to identify issues before submitting commits

### PR DESCRIPTION
**Pull request by k-doering-NOAA**
Addresses #23 .

### CODE REVIEW

**Comment by k-doering-NOAA** [comment]
Not quite ready to merge in - I am doing some testing.

**Comment by k-doering-NOAA** [comment]
@Bai-Li-NOAA I think this is now ready to merge in - note that the r cmd check will be failing, because one of the functions references the main branch, where the function does not yet exist. I tested with the gitleaks branch name and was able to get r cmd check to pass (a4734d5).

Note that the reusable workflow came from the NEFSC and SEFSC github SOPs.

File: .github/workflows/run-gitleaks.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+jobs:
+  gitleaks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: '0'
+      - name: gitleaks-action
+        uses: gitleaks/gitleaks-action@v1.6.0
```
Comment (Bai-Li-NOAA) [inline .github/workflows/run-gitleaks.yml:15]: Wonder if the action version can be updated to v2 based on the [example](https://github.com/gitleaks/gitleaks-action#usage-example) from gitleaks-action repo. Also, I am not sure if a env section is needed for providing GITHUB_TOKEN and GITLEAKS_LICENSE info. GITLEAKS_LICENSE is required for Organizations, not personal account.

File: README.md [hunk 2]
```diff
@@ -50,6 +50,11 @@ ghactions4r::use_doc_and_style_r(use_rm_dollar_sign = FALSE)
 ghactions4r::update_pkgdown()
 ```
 
+- To set up [gitleaks](https://github.com/zricethezav/gitleaks), which scans a project repository for accidentally committing secrets:
```
Comment (Bai-Li-NOAA) [inline README.md:53]: Maybe provide examples of secrets here? 

...secrets (e.g., passwords, api keys, and tokens in a repository).

**Review by Bai-Li-NOAA — APPROVED** [review]
Sorry, @k-doering-NOAA, I totally forgot about the task and thanks for setting up the gitleaks workflow. I provided a few minor suggestions in the review. 

Do you think there is a need to test the workflow by providing a file with a fake secret after the feature branch gets merged into the main branch? I am happy to set up a test if it is necessary.

**Comment by k-doering-NOAA** [comment]
Thanks @Bai-Li-NOAA, I think it would be great to test this out, as I haven't done any testing to confirm it can catch secrets. I will try to make changes to this tomorrow

**Comment by k-doering-NOAA** [comment]
I guess I should have read more about Gitleaks - it seems what the Github SOPs is suggesting is already out of date.

- Gitleaks is [mostly no longer free](https://blog.gitleaks.io/gitleaks-llc-announcement-d7d06a52e801). So, it seems we would all need a license to use it for secret scanning in an org. 
- It IS free for repos in personal accounts. It is not clear to me if it would work for personal accounts if they call a reusable workflow. We *could* test it.

@Bai-Li-NOAA , given this new information, do you think we should add it to our package? I'm a little hesitant to add something that won't work for the majority of folks without a license.

**Comment by k-doering-NOAA** [comment]
reading a bit more (https://github.com/gitleaks/gitleaks-action#i-really-need-a-secret-scanner-but-i-have-no-money-to-buy-a-license-what-can-i-do), it sounds like v1.6.0 is still free and will work, but apparently there are some "known issues".

**Comment by Bai-Li-NOAA** [comment]
@k-doering-NOAA, thanks for sharing the info. I guess I should have read the whole readme file from the gitleaks-action repo before making suggestions 😅. How about we discuss whether we really need a secret scanner or not during the coming modeling team meeting? In the meanwhile, we can always [pin the Gitleaks-Action yml to the last free version of Gitleaks-Action (v1.6.0)](https://github.com/gitleaks/gitleaks-action#i-really-need-a-secret-scanner-but-i-have-no-money-to-buy-a-license-what-can-i-do). "See here: [How to pin to v1.6.0](https://github.com/gitleaks/gitleaks-action/blob/master/v2.md#how-to-pin-to-v160)
Caveat: There are some known issues with that version, and it's no longer receiving updates. But it's better than nothing".

**Comment by k-doering-NOAA** [comment]
I think I will close this pull request for now, but keep the branch around. I think some NMFS level discussion of what to use is warranted, since the current recommendation to use gitleaks is outdated. I would only like to merge in what is necessary for NMFS developers so as to not confuse folks with additional options!

**Pull request closed without merging**